node/src/server-modules/GlideFtOptions.ts

```diff
@@ -118,3 +119,117 @@ export interface FtCreateOptions {
     /** The prefix of the key to be indexed. */
     prefixes?: GlideString[];
 }
+
+/** Additional parameters for {@link GlideFt.aggregate | FT.AGGREGATE} command. */
+export type FtAggregateOptions = {
+    /** Query timeout in milliseconds. */
+    timeout?: number;
+    /**
+     * {@link FtAggregateFilter | FILTER}, {@link FtAggregateLimit | LIMIT}, {@link FtAggregateGroupBy | GROUPBY},
+     * {@link FtAggregateSortBy | SORTBY} and {@link FtAggregateApply | APPLY} clauses, that can be repeated
+     * multiple times in any order and be freely intermixed. They are applied in the order specified,
+     * with the output of one clause feeding the input of the next clause.
+     */
+    clauses?: FtAggregateClause[];
+    /** The key/value pairs can be referenced from within the query expression. */
+    params?: [GlideString, GlideString][];
+} & (
+    | {
+          /** List of fields to load from the index. */
+          loadFields?: GlideString[];
+          /** `loadAll` and `loadFields` are mutually exclusive. */
+          loadAll?: never;
+      }
+    | {
+          /** Option to load all fields declared in the index */
+          loadAll?: boolean;
+          /** `loadAll` and `loadFields` are mutually exclusive. */
+          loadFields?: never;
+      }
+);
+
+/** The {@link GlideFt.aggregate | FT.AGGREGATE} clause type. */
+export enum FtAggregateClauseType {
+    LIMIT = "LIMIT",
+    FILTER = "FILTER",
+    GROUPBY = "GROUPBY",
+    SORTBY = "SORTBY",
+    APPLY = "APPLY",
+}
+
+export type FtAggregateClause =
+    | FtAggregateLimit
+    | FtAggregateFilter
+    | FtAggregateGroupBy
+    | FtAggregateSortBy
+    | FtAggregateApply;
+
+/** A clause for limiting the number of retained records. */
+export interface FtAggregateLimit {
+    type: FtAggregateClauseType.LIMIT;
```

Comment:
Instead of having `type` value present in each of the classes, why not have `toArgs()` function similar to JAVA and python inside this interface. Adding `toArgs()` to each interface and then calling `clauses[i].toArgs()` should solve this for us.

Ideally, the best solution will be to use class inheritance type solution, but converting all code classes might not be feasible, so, the above mentioned changes should work fine on interface as well.

This will do the following:
1. Reduce the code from the GlideFt.aggregate() function.
2. Remove the switch case block. It is a general best practice to avoid switch case because of its non extensibility.